### ISSUE TITLE
Country not available on the order edit screen if shop no longer sells/sends to the order country

### ISSUE THREAD

**Issue by webdados** [woothemes/woocommerce#41777]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Let's say the shop used to sell or send to 3 countries: Portugal, Spain, São Tomé e Príncipe.

Then the shop owner decides not to sell to France anymore, so he changes the settings on WooCommerce > General.

Then you go back to the order edit screen of a previous order sold to São Tomé e Príncipe and the original order country is not available on the dropdown and the first country available is selected, which means that if you save the order, you'll change it's country without even noticing it:
<img width="318" alt="Screenshot 2023-11-29 at 11 56 14" src="https://github.com/woocommerce/woocommerce/assets/6853605/79c290b9-ef9f-4d90-ad44-f7853a942f28">
<img width="313" alt="image" src="https://github.com/woocommerce/woocommerce/assets/6853605/872eea40-c8f2-4212-b5b5-7f423628b4de">
<img width="191" alt="image" src="https://github.com/woocommerce/woocommerce/assets/6853605/edda470b-b944-4836-829a-4bd1c1c74f71">

I went to that order and hit the update button without even touching the form, and the country changed to Portugal, as it was the first on the country select, which shouldn't have happened.

### Expected behavior

The country select on address edit forms on the order edit screen should contain:
- All the countries that apply (current order billing or shipping accepted countries)
- The original country set on the order

### Actual behavior

If a country is removed from the allowed countries to sell/send to, if you happen to hit the update button on a previous order, that was sold/sent to a non-allowed country at that time, you'll change the order country.

### Steps to reproduce

1. Allow WooCommerce to sell to countries A, B and C
2. Create an order to country C
3. Visit the order edit screen for that order: all should be OK
4. Go to the WooCommerce General settings and remove country C
5. Visit the order edit screen for that order again and hit update
6. The order country was now changed to A or B (depending what was the first on the country select field)

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): https://wordpress.local
Site address (URL): https://wordpress.local
WC Version: 8.4.0
REST API Version: ✔ 8.4.0
WC Blocks Version: ✔ 11.6.1
Action Scheduler Version: ✔ 3.7.0
Log Directory Writable: ✔
WP Version: ❌ 6.5-alpha-57137 . Há uma nova versão do WordPress disponível (6.5-alpha-57141)
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: –
Language: pt_PT
External object cache: –

### Server Environment ###

Server Info: nginx/1.16.0
PHP Version: 8.0.22
PHP Post Max Size: 1.000 MB
PHP Time Limit: 1200
PHP Max Input Vars: 4000
cURL Version: 7.84.0
(SecureTransport) OpenSSL/1.1.1q

SUHOSIN Installed: –
MySQL Version: 8.0.16
Max Upload Size: 300 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 8.4.0
WC Database Prefix: wp_
Tamanho total da base de dados: 3831.47MB
Tamanho dos dados da base de dados: 3775.40MB
Tamanho do índice da base de dados: 56.07MB
wp_woocommerce_sessions: Dados: 1.45MB + Índice: 0.06MB + Engine InnoDB
wp_woocommerce_api_keys: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Dados: 0.16MB + Índice: 0.06MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Dados: 1.52MB + Índice: 0.77MB + Engine InnoDB
wp_woocommerce_tax_rates: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_woocommerce_shipping_zones: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_woocommerce_log: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Dados: 2.02MB + Índice: 0.73MB + Engine InnoDB
wp_actionscheduler_claims: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Dados: 1.50MB + Índice: 0.44MB + Engine InnoDB
wp_aelia_dismissed_messages: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_aelia_exchange_rates_history: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_activity: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_bank_transfer_log: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_email_templates: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_entries: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_fail_attempts: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_form_field: Dados: 0.06MB + Índice: 0.00MB + Engine InnoDB
wp_arm_forms: Dados: 0.13MB + Índice: 0.00MB + Engine InnoDB
wp_arm_lockdown: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_login_history: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_member_templates: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_members: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_arm_membership_setup: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_payment_log: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_subscription_plans: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_arm_termmeta: Dados: 0.06MB + Índice: 0.06MB + Engine InnoDB
wp_automatewoo_abandoned_carts: Dados: 0.02MB + Índice: 0.08MB + Engine InnoDB
wp_automatewoo_customer_meta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_automatewoo_customers: Dados: 0.02MB + Índice: 0.09MB + Engine InnoDB
wp_automatewoo_events: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_automatewoo_guest_meta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_automatewoo_guests: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_automatewoo_log_meta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_automatewoo_logs: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_automatewoo_queue: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_automatewoo_queue_meta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_automatewoo_unsubscribes: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_berocket_termmeta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_chefpanda_restrictions: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_commentmeta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_comments: Dados: 0.41MB + Índice: 0.38MB + Engine InnoDB
wp_CTT: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_CTT_marketplace_order: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_egoi_form_subscribers: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_egoi_map_fields: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_et_bloom_stats: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_eupago_mbway: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_ewwwio_images: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_facetwp_cache: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_facetwp_index: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_facetwp_temp: Dados: 2.52MB + Índice: 4.45MB + Engine InnoDB
wp_favethemes_currency_converter: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_fcc_currencies: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_font_color: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_frm_fields: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_frm_forms: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_frm_item_metas: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_frm_items: Dados: 0.02MB + Índice: 0.08MB + Engine InnoDB
wp_gp_glossaries: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_gp_glossary_entries: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_gp_meta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_gp_originals: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_gp_permissions: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_gp_projects: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_gp_translation_sets: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_gp_translations: Dados: 0.02MB + Índice: 0.11MB + Engine InnoDB
wp_homey_earnings: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_homey_fields_builder: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_homey_map: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_homey_payouts: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_homey_thread_messages: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_homey_threads: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_ibwp_mp_submission: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_background_task: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_content_status: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_icl_core_status: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_icl_flags: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_icl_languages: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_icl_languages_translations: Dados: 0.19MB + Índice: 0.11MB + Engine InnoDB
wp_icl_locale_map: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_message_status: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_icl_mo_files_domains: Dados: 0.05MB + Índice: 0.02MB + Engine InnoDB
wp_icl_node: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_reminders: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_string_batches: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_string_packages: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_string_pages: Dados: 0.27MB + Índice: 0.39MB + Engine InnoDB
wp_icl_string_positions: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_icl_string_status: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_icl_string_translations: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_icl_string_urls: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_icl_strings: Dados: 0.19MB + Índice: 0.38MB + Engine InnoDB
wp_icl_translate: Dados: 0.05MB + Índice: 0.02MB + Engine InnoDB
wp_icl_translate_job: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_icl_translation_batches: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_translation_downloads: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_icl_translation_status: Dados: 0.05MB + Índice: 0.02MB + Engine InnoDB
wp_icl_translations: Dados: 1.52MB + Índice: 3.77MB + Engine InnoDB
wp_itsec_lockouts: Dados: 0.02MB + Índice: 0.08MB + Engine InnoDB
wp_itsec_log: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_itsec_temp: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_ix_log: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_jckwds: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_links: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_mipconnector_attribute_group_map: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_mipconnector_attribute_map: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_mipconnector_brand_map: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_mipconnector_brand_plugin_map: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_mipconnector_cache: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_mipconnector_category_map: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_mipconnector_file_log: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_mipconnector_image_url: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_import_process_attribute: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_import_process_attribute_group: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_import_process_brand: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_import_process_brand_plugin: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_import_process_category: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_import_process_product: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_import_process_tag: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_import_process_variation: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_order_log: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_product_image_url: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_mipconnector_product_map: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_mipconnector_product_url: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_mipconnector_reference_data: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_mipconnector_shipping_service: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_mipconnector_tag_map: Dados: 0.02MB + Índice: 0.09MB + Engine InnoDB
wp_mipconnector_variation_map: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_options: Dados: 3679.25MB + Índice: 0.30MB + Engine InnoDB
wp_pmxe_exports: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_pmxe_google_cats: Dados: 0.39MB + Índice: 0.00MB + Engine InnoDB
wp_pmxe_posts: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_pmxe_templates: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_pmxi_files: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_pmxi_history: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_pmxi_images: Dados: 0.05MB + Índice: 0.00MB + Engine InnoDB
wp_pmxi_imports: Dados: 0.28MB + Índice: 0.00MB + Engine InnoDB
wp_pmxi_posts: Dados: 1.52MB + Índice: 0.38MB + Engine InnoDB
wp_pmxi_templates: Dados: 0.14MB + Índice: 0.00MB + Engine InnoDB
wp_po_plugins: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_portuguese_postcodes: Dados: 21.55MB + Índice: 0.00MB + Engine InnoDB
wp_postmeta: Dados: 30.56MB + Índice: 24.06MB + Engine InnoDB
wp_posts: Dados: 5.52MB + Índice: 3.73MB + Engine InnoDB
wp_psmsc_agents: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_attachments: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_categories: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_custom_fields: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_customers: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_email_notifications: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_email_otp: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_holidays: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_logs: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_options: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_priorities: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_statuses: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_threads: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_tickets: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_wh_exceptions: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_psmsc_working_hrs: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_rank_math_404_logs: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_rank_math_internal_links: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_rank_math_internal_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_rank_math_redirections: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_rank_math_redirections_cache: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_rank_math_sc_analytics: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_seur_custom_rates: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_seur_svpr: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_shiptimize: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_shiptimize_marketplace_order: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sibs_payment_information: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sibs_transaction: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_social_users: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_sp_admins: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_answers: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_config: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sp_exceptions: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sp_hooks_admins_multi: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_hooks_customers: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_hooks_templates: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_hooks_unicode: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_marketing_filter: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sp_optout_deleted: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_optout_prices: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_optout_texts: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_outofstock: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_ownnumbersenderIDs: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_routes: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_settings: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_sms_history: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_smsuser: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sp_template: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sp_textsenderIDs: Dados: 0.00MB + Índice: 0.00MB + Engine MyISAM
wp_sti_atos_eleitorais: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sti_distritos: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sti_entidades: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sti_locais: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sti_temp_associados: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_sti_temp_associados_exist: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_term_relationships: Dados: 1.52MB + Índice: 1.52MB + Engine InnoDB
wp_term_taxonomy: Dados: 0.06MB + Índice: 0.06MB + Engine InnoDB
wp_termmeta: Dados: 0.09MB + Índice: 0.11MB + Engine InnoDB
wp_terms: Dados: 0.06MB + Índice: 0.03MB + Engine InnoDB
wp_tm_taskmeta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_tm_tasks: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_usermeta: Dados: 0.09MB + Índice: 0.09MB + Engine InnoDB
wp_users: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_wc_admin_note_actions: Dados: 0.06MB + Índice: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Dados: 0.09MB + Índice: 0.00MB + Engine InnoDB
wp_wc_category_lookup: Dados: 0.05MB + Índice: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_wc_deposits_payment_plans: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wc_deposits_payment_plans_schedule: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wc_download_log: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_wc_ie_scheduled_docs: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wc_order_addresses: Dados: 0.14MB + Índice: 0.19MB + Engine InnoDB
wp_wc_order_coupon_lookup: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_wc_order_operational_data: Dados: 0.09MB + Índice: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Dados: 0.08MB + Índice: 0.06MB + Engine InnoDB
wp_wc_order_stats: Dados: 0.08MB + Índice: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Dados: 0.05MB + Índice: 0.03MB + Engine InnoDB
wp_wc_orders: Dados: 0.27MB + Índice: 0.31MB + Engine InnoDB
wp_wc_orders_meta: Dados: 1.52MB + Índice: 2.95MB + Engine InnoDB
wp_wc_product_attributes_lookup: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Dados: 1.52MB + Índice: 1.28MB + Engine InnoDB
wp_wc_rate_limits: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wc_revolut_orders: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_wc_tax_rate_classes: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wc_webhooks: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wcfm_daily_analysis: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wcfm_detailed_analysis: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_enquiries: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_enquiries_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_enquiries_response: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_enquiries_response_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_fbc_chat_rows: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_fbc_chat_sessions: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wcfm_fbc_chat_visitors: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_fbc_offline_messages: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_following_followers: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_orders: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wcfm_marketplace_orders_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_product_multivendor: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wcfm_marketplace_refund_request: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_refund_request_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_reverse_withdrawal: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_reverse_withdrawal_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_review_rating_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_reviews: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_reviews_response: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_reviews_response_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_shipping_zone_locations: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_shipping_zone_methods: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_store_taxonomies: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_vendor_ledger: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_withdraw_request: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_marketplace_withdraw_request_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_messages: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_messages_modifier: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_messages_stat: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_support: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_support_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_support_response: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcfm_support_response_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wcpdf_invoice_number: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wdr_order_discounts: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wdr_order_item_discounts: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wdr_rules: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_woo_dpd_pickup_points: Dados: 2.52MB + Índice: 0.27MB + Engine InnoDB
wp_woo_wallet_transaction_meta: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_woo_wallet_transactions: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_woocommerce_api_manager_secure_hash: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_woocommerce_oppcw_customer_contexts: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_woocommerce_oppcw_documents: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_woocommerce_oppcw_ecc: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_woocommerce_oppcw_schema_version: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_woocommerce_oppcw_storage: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_woocommerce_oppcw_transactions: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_table_rates: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wpbdp_fees: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wpbdp_form_fields: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wpbdp_listing_fees: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_wpbdp_payments: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_wpbdp_payments_items: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wpbdp_submit_state: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wpmailsmtp_debug_events: Dados: 0.06MB + Índice: 0.00MB + Engine InnoDB
wp_wpmailsmtp_tasks_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wpmudev_chat_log: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_wpmudev_chat_message: Dados: 0.02MB + Índice: 0.11MB + Engine InnoDB
wp_wpmudev_chat_users: Dados: 0.02MB + Índice: 0.06MB + Engine InnoDB
wp_wps_hit: Dados: 0.02MB + Índice: 0.14MB + Engine InnoDB
wp_wps_index: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_wps_key: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wps_object_term: Dados: 0.02MB + Índice: 0.09MB + Engine InnoDB
wp_wps_object_type: Dados: 0.02MB + Índice: 0.08MB + Engine InnoDB
wp_wps_query: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wps_uri: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wps_user_agent: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_wpvivid_scan_result: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_wpvivid_unused_uploads_files: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB
wp_yith_wcwl: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_yith_wcwl_lists: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_yoast_indexable: Dados: 10.52MB + Índice: 4.42MB + Engine InnoDB
wp_yoast_indexable_hierarchy: Dados: 0.02MB + Índice: 0.05MB + Engine InnoDB
wp_yoast_migrations: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_yoast_primary_term: Dados: 0.02MB + Índice: 0.03MB + Engine InnoDB
wp_yoast_seo_links: Dados: 0.02MB + Índice: 0.02MB + Engine InnoDB
wp_yoast_seo_meta: Dados: 0.02MB + Índice: 0.00MB + Engine InnoDB

### Post Type Counts ###

acf-field: 19
acf-field-group: 3
amn_smtp: 1
atividade: 4
attachment: 6281
aw_workflow: 2
blocks: 8
chefpanda_market: 3
easy-pricing-table: 1
featured_item: 8
frm_form_actions: 1
frm_styles: 1
homey_reservation: 2
ibwp_mp_popup: 1
listing: 1
mb_payment_request: 1
nav_menu_item: 27
oksofasstore: 3
page: 59
person: 5
post: 25
private_post: 4
product: 7954
product_variation: 131
revision: 189
shop_coupon: 18
shop_order: 539
shop_order_refund: 89
shop_subscription: 9
szbdzones: 1
was: 1
wc_conditional_fee: 1
wc_order_status: 9
wc-donation: 2
wc-donation-report: 5
wcdp_payment: 4
wd_client_contract: 1
wd_support_ticket: 1
webdados_client_serv: 10
webdados_service: 8
wishlist: 1
wp_block: 1
wp_global_styles: 4
wp_navigation: 1
wp_template: 2
wpbdp_listing: 1
wpcf7_contact_form: 6
wps_subscriptions: 1
wpsc_ticket_thread: 3

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ❌As mensagens de erro não devem ser exibidas aos visitantes.

### Active Plugins (7) ###

Multibanco, MB WAY, Cartão de crédito e Payshop (IfthenPay) para WooCommerce: por PT Woo Plugins (by Webdados) – 8.8.0
Plugin Check: por Plugin Review Team – 0.2.1
Distritos de Portugal para WooCommerce: por PT Woo Plugins (by Webdados) – 3.4.1
WooCommerce Beta Tester: por WooCommerce – 2.2.4
WooCommerce: por Automattic – 8.4.0-beta.1
WordPress Beta Tester: por WordPress Upgrade/Install Team – 3.5.5
WP Mail SMTP: por WP Mail SMTP – 3.10.0

### Inactive Plugins (293) ###

Not relevant

### Dropin Plugins (1) ###

db.php: db.php

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: EUR (€)
Currency Position: right_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to Woo.com: –
Enforce Approved Product Download Directories: –
HPOS feature screen enabled: –
HPOS feature enabled: –
Order datastore: WC_Order_Data_Store_CPT
HPOS data sync enabled: ✔

### WC Pages ###

Página da Loja: #14857 - /loja-pt/
Carrinho: #14 - /cart/
Finalizar compras: #16786 - /checkout-2/
A minha conta: #16787 - /my-account-2/
Termos e condições: ❌ Página não definida

### Theme ###

Name: Storefront Child
Version: 20230130.01
Author URL: 
Child Theme: ✔
Parent Theme Name: Storefront
Parent Theme Version: 4.5.3
Parent Theme Author URL: https://woocommerce.com/
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-variation-management
product-virtual-downloadable
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: customize-store
minified-js
new-product-management-experience
product-external-affiliate
settings
async-product-editor-category-field
beta-tester-slotfill-examples

Daily Cron: ✔ Next scheduled: 2023-11-22 15:19:07 +00:00
Options: ✔
Notes: 169
Onboarding: completed

### Action Scheduler ###

Concluída: 127
Oldest: 2023-10-30 10:30:48 +0000
Newest: 2023-11-29 12:16:17 +0000

Falhada: 2.311
Oldest: 2023-02-01 17:31:32 +0000
Newest: 2023-11-22 17:04:17 +0000

Pendente: 2
Oldest: 2023-11-30 12:07:54 +0000
Newest: 2023-11-30 12:07:54 +0000

### Status report information ###

Generated at: 2023-11-29 12:28:02 +00:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by webdados** [comment]
Currently the countries for the select fields are fetched with [`WC()->countries->get_allowed_countries()`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-order-data.php#L84) and [`WC()->countries->get_shipping_countries()`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/admin/meta-boxes/class-wc-meta-box-order-data.php#L143) in class-wc-meta-box-order-data.php

I think a good approach would be to have `WC_Order` methods `get_allowed_countries()` and `get_allowed_shipping_countries()`, that would first call `WC()->countries->get_allowed_countries()` and `WC()->countries->get_shipping_countries()` and then, if the current order billing/shipping country is not present in the array, add it to it before returning the countries list.

Another approach could be to have new `WC_Countries` methods `get_allowed_countries_for_order( $order_id )` and `get_shipping_countries_for_order( $order_id )` with the same behaviour as mentioned above.

Please discuss the best way to approach this, and if you agree in one of these solutions I can submit a PR.

**Comment by jorgeatorres** [comment]
Hi @webdados!

Thanks for reaching out. This does look like an omission on our part and we definitely welcome a PR! As for your question, we feel this belongs to the order class itself, and not `WC_Countries` (as this has remained order-agnostic so far).
We think that it should be made clear that the new methods are to be used when "editing" an order and not necessarily replace existing APIs. For example, using names like `get_allowed_countries_for_edit()` or similar will help us convey that.

Let us know if you have any further questions. Look forward to your contribution!

**Comment by webdados** [comment]
Depending on the feedback from the just issued PR, I will do the same of the customer billing and shipping address edit forms, both in the backend and frontend, as the exact same problem exists.

**Comment by flakas** [comment]
Hi @webdados - thanks for reporting this, and for the detail you gave.

I know this has been open for a long time, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

It looks like this has already been fixed via #41777, or it's no longer happening on trunk.

I'm going to close this one. If you're still facing this issue on a current version, please reply with steps to reproduce and we'll happily take another look.

**Comment by webdados** [comment]
@flakas That PR was never finished or merged, but i can confirm the problem no longer exists in trunk.

**Comment by flakas** [comment]
Excellent, thank you for confirming!
